server/src/main/java/org/opensearch/action/search/SearchRequestContext.java

```diff
@@ -23,20 +37,29 @@
  */
 @InternalApi
 public class SearchRequestContext {
+    private static final Logger logger = LogManager.getLogger();
     private final SearchRequestOperationsListener searchRequestOperationsListener;
     private long absoluteStartNanos;
     private final Map<String, Long> phaseTookMap;
     private TotalHits totalHits;
     private final EnumMap<ShardStatsFieldNames, Integer> shardStats;
 
     private final SearchRequest searchRequest;
-
-    SearchRequestContext(final SearchRequestOperationsListener searchRequestOperationsListener, final SearchRequest searchRequest) {
+    private final List<TaskResourceInfo> phaseResourceUsage;
+    private final Supplier<ThreadContext> threadContextSupplier;
+
+    SearchRequestContext(
+        final SearchRequestOperationsListener searchRequestOperationsListener,
+        final SearchRequest searchRequest,
+        final Supplier<ThreadContext> threadContextSupplier
```

Comment:
We need to refactor the `SearchRequestContext` class with builder pattern. Also, some parameters added in `TransportSearchAction` like tracer should in theory belong to this context.